sentry-spring-boot-starter/src/main/java/io/sentry/spring/boot/SentryAutoConfiguration.java

```diff
@@ -125,6 +125,53 @@
     @Open
     static class SentryWebMvcConfiguration {
 
+      @Bean
+      @ConditionalOnClass(SecurityContextHolder.class)
+      @Order(1)
+      public SpringSecuritySentryUserProvider springSecuritySentryUserProvider(
+          final @NotNull IHub hub) {
+        return new SpringSecuritySentryUserProvider(hub);
+      }
+
+      /**
+       * Configures {@link SentryUserFilter}. By default it runs as the last filter in order to make
+       * sure that all potential authentication information is propagated to {@link
+       * HttpServletRequest#getUserPrincipal()}. If Spring Security is auto-configured, its order is
+       * set to run after Spring Security.
+       *
+       * @param hub the Sentry hub
+       * @param sentryProperties the Sentry properties
+       * @param sentryUserProvider the user provider
+       * @param securityProperties the Spring Security properties
+       * @return {@link SentryUserFilter} registration bean
+       */
+      @Bean
+      @ConditionalOnBean(SentryUserProvider.class)
+      FilterRegistrationBean<SentryUserFilter> sentryUserFilter(
+          final @NotNull IHub hub,
+          final @NotNull SentryProperties sentryProperties,
+          final @NotNull List<SentryUserProvider> sentryUserProvider,
+          final @NotNull Optional<SecurityProperties> securityProperties) {
+        final FilterRegistrationBean<SentryUserFilter> filter = new FilterRegistrationBean<>();
+        filter.setFilter(new SentryUserFilter(hub, sentryUserProvider));
+        filter.setOrder(resolveUserFilterOrder(sentryProperties, securityProperties));
+        return filter;
+      }
+
+      private @NotNull Integer resolveUserFilterOrder(
+          final @NotNull SentryProperties sentryProperties,
+          final @NotNull Optional<SecurityProperties> securityProperties) {
+        return Optional.ofNullable(sentryProperties.getUserFilterOrder())
+            .orElseGet(
+                () ->
+                    securityProperties
+                        .map(
+                            props ->
+                                props.getFilter().getOrder()
+                                    + 10) // filter should run after security filters
```

Comment:
mm, is it a magic number? how can we improve that?

Reply:
If you insist we can extract it to a constant but not sure how to name it. Filter order is defined with magic numbers in Spring Security (https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/security/SecurityProperties.java). Here we make sure that the order is higher, but we leave place for extra filters to be added.

We can also leave out this option, use only `LOWEST_PRECEDENCE` and give users a way to overwrite complete filter definition.

Reply:
Removed this option and left LOWEST_PRECEDENCE since this will cover 99% of cases anyways. If users need to customize the order, they can set a value to `sentry.user-filter-order`